warehouse/forklift/legacy.py

```diff
@@ -570,6 +622,12 @@ class MetadataForm(forms.Form):
         description="Requires-Dist",
         validators=[wtforms.validators.Optional(), _validate_legacy_dist_req_list],
     )
+    provides_extra = (
+        ListField(
+            description="Provides-Extra",
+            validators=[wtforms.validators.Optional(), _validate_extras],
+        ),
+    )
```

Comment:
Ditto here about adding this to the model: https://github.com/pypi/warehouse/blob/main/warehouse/packaging/models.py#L423-L604